includes/discovery/sensors/dbm/fabos.inc.php

```diff
@@ -1,70 +1,45 @@
 <?php
 
+use LibreNMS\Util\Oid;
 use LibreNMS\Util\Rewrite;
 
-$fabosSfpRxPower = snmpwalk_array_num($device, '.1.3.6.1.4.1.1588.2.1.1.1.28.1.1.4'); // FA-EXT-MIB::swSfpRxPower
-$fabosSfpTxPower = snmpwalk_array_num($device, '.1.3.6.1.4.1.1588.2.1.1.1.28.1.1.5'); // FA-EXT-MIB::swSfpTxPower
-if (! empty($fabosSfpRxPower) || ! empty($fabosSfpTxPower)) {
-    $ifDescr = snmpwalk_group($device, 'ifDescr', 'IF-MIB', 0)['ifDescr'] ?? [];
-    $ifAdminStatus = snmpwalk_group($device, 'ifAdminStatus', 'IF-MIB', 0)['ifAdminStatus'] ?? [];
-}
+$fabosSfpRxPower = SnmpQuery::hideMib()->mibs(['FA-EXT-MIB'])->numericIndex()->walk('FA-EXT-MIB::swSfpRxPower')->valuesByIndex();
+$fabosSfpTxPower = SnmpQuery::hideMib()->mibs(['FA-EXT-MIB'])->numericIndex()->walk('FA-EXT-MIB::swSfpTxPower')->valuesByIndex();
 
-foreach ($fabosSfpRxPower as $oid => $entry) {
-    foreach ($entry as $index => $current) {
-        if (is_numeric($current)) {
-            $ifIndex = $index + 1073741823;
-            if ($ifAdminStatus[$ifIndex] == '1') {
-                discover_sensor(
-                    null,
-                    'dbm',
-                    $device,
-                    ".$oid.$index",
-                    'swSfpRxPower.' . $index,
-                    'brocade',
-                    Rewrite::shortenIfName($ifDescr[$ifIndex]) . ' RX',
-                    1,
-                    1,
-                    -5,
-                    null,
-                    null,
-                    0,
-                    $current,
-                    'snmp',
-                    $ifIndex,
-                    'ports',
-                    null,
-                    'Receive Power'
-                );
-            }
-        }
-    }
+$fabosSfpPower = array_merge_recursive($fabosSfpTxPower, $fabosSfpRxPower);
+
+if (! empty($fabosSfpPower)) {
+    $ifDescr = SnmpQuery::hideMib()->mibs(['IF-MIB'])->walk('IF-MIB::ifDescr')->table(1) ?? [];
+    $ifAdminStatus = SnmpQuery::hideMib()->mibs(['IF-MIB'])->walk('IF-MIB::ifAdminStatus')->table(1) ?? [];
 }
 
-foreach ($fabosSfpTxPower as $oid => $entry) {
-    foreach ($entry as $index => $current) {
+foreach ($fabosSfpPower as $fullIndex => $entry) {
+    foreach ($entry as $oid => $current) {
         if (is_numeric($current)) {
+            $index = array_slice(explode('.', $fullIndex), -1)[0];
             $ifIndex = $index + 1073741823;
```

Comment:
I am very dubious about this.  I know it is in the legacy code, but it seems unlikely to be correct in all situations.

Reply:
Are you sure we don't have something like `ENTITY-MIB::entAliasMappingIdentifier`

Reply:
`$index` is now renamed `$connUnitPortIndex` and the conversion between `$ifIndex` and `$connUnitPortIndex` is indeed this constant. This is the only mapping in the snmp replies of the device. I checked with a few greps on the `$ifIndex` and it only appears in IF-MIB.

Reply:
The only other option I found is `ifAlias` and `connUnitPortName` with a string comparison. Would you prefer that instead of the constant ? A little bit less efficient in term of compute, but probably a little bit safer.

Reply:
Sensors already contains the port_id. So that part is OK. 
But to map the connUnitXXXX to the sensor, I need to have it mapped to an ifIndex.

So the 2 options are :
- Current PR implementation
- Or connUnitPortName string-mapped to ifAlias

Do you prefer the 2nd one ?

Reply:
If both tables use connUnitPortIndex, why not just set that to the sensor's `entPhysicalIndex` and transceivers `entity_physical_index`?

Reply:
Good. 
Then I'll check about how I can rewrite the discovery for both sensors AND transceivers, make it as good as possible, and make it visible how the constant is computed. As I only have pizzabox standalone switches, I won't be able to test anything else but will be ready if we get feedback from chassis users.